util/cinc-install.sh

```diff
@@ -0,0 +1,824 @@
+      platform_version="6"
+    else
+      platform_version=$major_version
+    fi
+    ;;
+  "freebsd")
+    platform_version=$major_version
+    ;;
+  "sles")
+    platform_version=$major_version
+    ;;
+  "opensuseleap")
+    platform_version=$major_version
+    ;;
+esac
+
+# normalize the architecture we detected
+case $machine in
+  "x86_64"|"amd64"|"x64")
+    machine="x86_64"
+    ;;
+  "i386"|"i86pc"|"x86"|"i686")
+    machine="i386"
+    ;;
+  "sparc"|"sun4u"|"sun4v")
+    machine="sparc"
+    ;;
+esac
+
+if test "$platform" = "ubuntu"; then
+  case $machine in
+    "arm64"|"aarch64")
+      machine="arm64"
+      ;;
+    "x86_64"|"amd64"|"x64")
+      machine="amd64"
+      ;;
+    "i386"|"i86pc"|"x86"|"i686")
+      machine="i386"
+      ;;
+  esac
+fi
+
+if test "x$platform_version" = "x"; then
+  echo "Unable to determine platform version!"
+  report_bug
+  exit 1
+fi
+
+if test "x$platform" = "xsolaris2"; then
+  # hack up the path on Solaris to find wget, pkgadd
+  PATH=/usr/sfw/bin:/usr/sbin:$PATH
+  export PATH
+fi
+
+# Region detection
+instance_metadata_file=$tmp_dir/instance_metadata
+do_download "http://169.254.169.254/latest/dynamic/instance-identity/document" ${instance_metadata_file}
+region=`cat ${instance_metadata_file} | awk '$1 =="\"region\"" {print $3}' | sed 's/"//g; s/,//g'`
+
+# Download domain detection
+if [ "${region}" != "${region#cn-*}" ]
+then
+  download_domain="amazonaws.com.cn"
+else
+  download_domain="amazonaws.com"
+fi
+
+echo "${platform} ${platform_version} ${machine} ${region}"
+
+############
+# end of platform_detection.sh
+############
+
+
+# All of the download utilities in this script load common proxy env vars.
+# If variables are set they will override any existing env vars.
+# Otherwise, default proxy env vars will be loaded by the respective
+# download utility.
+
+if test "x$https_proxy" != "x"; then
+  echo "setting https_proxy: $https_proxy"
+  HTTPS_PROXY=$https_proxy
+  https_proxy=$https_proxy
+  export HTTPS_PROXY
+  export https_proxy
+fi
+
+if test "x$http_proxy" != "x"; then
+  echo "setting http_proxy: $http_proxy"
+  HTTP_PROXY=$http_proxy
+  http_proxy=$http_proxy
+  export HTTP_PROXY
+  export http_proxy
+fi
+
+if test "x$ftp_proxy" != "x"; then
+  echo "setting ftp_proxy: $ftp_proxy"
+  FTP_PROXY=$ftp_proxy
+  ftp_proxy=$ftp_proxy
+  export FTP_PROXY
+  export ftp_proxy
+fi
+
+if test "x$no_proxy" != "x"; then
+  echo "setting no_proxy: $no_proxy"
+  NO_PROXY=$no_proxy
+  no_proxy=$no_proxy
+  export NO_PROXY
+  export no_proxy
+fi
+
+
+# create_download_url.sh
+############
+# This section creates the url of the package to download.
+#
+# Inputs:
+# $channel:
+# $project:
+# $version:
+# $platform:
+# $platform_version:
+# $machine:
+# $tmp_dir:
+#
+# Outputs:
+# $download_url:
+# $sha256:
+############
+
+bucket_url="https://${region}-aws-parallelcluster.s3.${region}.${download_domain}/archives/${project}/${platform}/${platform_version}/"
```

Comment:
because all the. Ubuntu packages have the same name but a different footprint. The alternative was to rename the packages, but I preferred to have a different tree